analysis/src/pipeline.rs

```diff
@@ -40,57 +38,62 @@ where
-                                let f = Function::with_uuid(name, uuid.clone(), region.name().clone());
-                                Some((entry, f))
+                        |ct| match ct {
+                            &CallTarget::Todo(Rvalue::Constant { value: entry, .. }, ref maybe_name, ref uuid) => {
+                                finished_functions.insert(entry);
+                                match Function::with_uuid::<A>(entry, uuid, &region, maybe_name.clone(), config.clone()) {
+                                    Ok(mut f) => {
+                                        let addresses = f.collect_call_addresses();
+                                        targets.extend_from_slice(&addresses);
+                                        let _ = ssa_convertion(&mut f);
+                                        Some(f)
+                                    },
+                                    Err(e) => { failures.push((entry, e)); None },
+                                }
                             }
-                            Some(_) => None,
-                            None => unreachable!(),
+                            _ => None,
                         }
-                    )
-            );
-
-            while !targets.is_empty() {
-                info!("disassemble {:?}", targets);
-                let new_targets: Vec<Vec<(u64, Function)>> = targets
-                    .into_iter()
-                    .map(
-                        |(entry, f)| {
-                            let tx = tx.clone();
-                            let mut f = Function::disassemble::<A>(Some(f), config.clone(), &region, entry);
-                            f.entry_point = f.find_basic_block_by_start(entry);
-                            let new_ct = f.collect_calls()
-                                .into_iter()
-                                .filter_map(
-                                    |rv| {
-                                        if let Rvalue::Constant { value, .. } = rv {
-                                            if !functions.contains_key(&value) && entry != value {
-                                                return Some((value, Function::new(format!("func_0x{:x}", value), region.name().clone())));
-                                            }
-                                        }
-                                        None
-                                    }
-                                )
-                                .collect::<Vec<(u64, Function)>>();
+                    ).collect::<Vec<Function>>();
 
-                            let _ = ssa_convertion(&mut f);
+            // now we send the first burst of functions;
+            // stop unwrapping;
+            {
+                let tx = tx.clone();
+                tx.send_all(stream::iter(functions.into_iter().map(|f| Ok(f)))).wait().unwrap().0;
```

Comment:
This will block for a UI component; I have this fixed in another PR i'll submit which is rebased on this, but I don't feel like backporting the commit since it's already fixed.

Also, @flanfly I think we should consider returning UUIDs instead of Functions.

The reason for this is, but mainly:

I need to re-insert the functions back into program in CLI to fix a few bugs that's too long to explain now; this would be solved if pipeline did the insertion, then returned the UUID (it will probably be faster too is my guess), which I could use to lookup, ignore, etc., since they're already present in program, where they should be.